### PR TITLE
DAOS-13214 engine: Setup modules in DRPC SetUp handler

### PR DESCRIPTION
**Pull request by tanabarr**
Setup all modules from within the dRPC setup handler to ensure this is
performed before the ready state is set in the control-plane.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by wangshilong — APPROVED** [review]
Looks reasonable, but missed Test-nvme: auto_md_on_ssd

**Review by NiuYawei — APPROVED** [review]

**Comment by tanabarr** [comment]
> Looks reasonable, but missed Test-nvme: auto_md_on_ssd

Why should this pragma be used for this PR? To me it doesn't seem like MD-on-SSD code.

File: src/include/daos_srv/daos_engine.h [hunk 1]
```diff
@@ -796,9 +796,11 @@ enum dss_media_error_type {
...
 void dss_init_state_set(enum dss_init_state state);
 
-/* Notify control-plane of a bio error. */
-int
-ds_notify_bio_error(int media_err_type, int tgt_id);
+/** Call module setup from drpc setup call handler. */
+int dss_module_setup_all(void);
+
+/** Notify control-plane of a bio error. */
+int ds_notify_bio_error(int media_err_type, int tgt_id);
```
Comment (ashleypittman) [inline src/include/daos_srv/daos_engine.h:803]: Note our coding style is and always has been the below.  We've just not been very good at sticking to it.  No need to re-push here but please consider this in future.
```suggestion
/** Call module setup from drpc setup call handler. */
int
dss_module_setup_all(void);

/** Notify control-plane of a bio error. */
int
ds_notify_bio_error(int media_err_type, int tgt_id);
```

Comment (tanabarr) [reply src/include/daos_srv/daos_engine.h:803]: Thanks @ashleypittman , I was trying to keep consistency with the code in close proximity to the addition but I will stick to the style prescribed.

**Comment by tanabarr** [comment]
clang-format failure seems okay with Ashley, Jira Report check seems stuck, Both are not required so can we please force land this?

**Comment by wangshilong** [comment]
> > Looks reasonable, but missed Test-nvme: auto_md_on_ssd
> 
> Why should this pragma be used for this PR? To me it doesn't seem like MD-on-SSD code.

 I supposed recreate_pooltgts() was called during setup for MD-on-SSD, i am fine without tag, but at least restart DAOS could work fine for MD-on-SSD should be verified locally at least.

**Comment by tanabarr** [comment]
> > > Looks reasonable, but missed Test-nvme: auto_md_on_ssd
> > 
> > 
> > Why should this pragma be used for this PR? To me it doesn't seem like MD-on-SSD code.
> 
> I supposed recreate_pooltgts() was called during setup for MD-on-SSD, i am fine without tag, but at least restart DAOS could work fine for MD-on-SSD should be verified locally at least.

I have verified simple restart case with `dmg system stop/start` with a single rank on this PR and it looks good.

**Comment by tanabarr** [comment]
@daos-stack/daos-gatekeeper can this PR please be landed

**Pull request merged**